hazelcast/src/test/java/com/hazelcast/test/starter/constructor/HazelcastExpiryPolicyConstructor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.test.starter.constructor;
+
+import com.hazelcast.test.starter.HazelcastStarterConstructor;
+
+import java.lang.reflect.Constructor;
+
+import static com.hazelcast.test.starter.ReflectionUtils.getFieldValueReflectively;
+
+@HazelcastStarterConstructor(classNames = {"com.hazelcast.cache.HazelcastExpiryPolicy"})
+public class HazelcastExpiryPolicyConstructor extends AbstractStarterObjectConstructor {
+
+    public HazelcastExpiryPolicyConstructor(Class<?> targetClass) {
+        super(targetClass);
+    }
+
+    @Override
+    Object createNew0(Object delegate) throws Exception {
+        Constructor<?> constructor = targetClass.getConstructor(Long.TYPE, Long.TYPE, Long.TYPE);
+
+        Object create = getFieldValueReflectively(delegate, "create");
```

Comment:
```suggestion
        Duration create = getFieldValueReflectively(delegate, "create");
        Duration access = getFieldValueReflectively(delegate, "access");
        Duration update = getFieldValueReflectively(delegate, "update");

        Object[] args = new Object[]{
                create.getTimeUnit().toMillis(create.getDurationAmount()),
                access.getTimeUnit().toMillis(access.getDurationAmount()),
                update.getTimeUnit().toMillis(update.getDurationAmount()),
        };
```

`Duration` is a `javax.cache` API class so it will not be loaded by the `HazelcastAPIDelegatingClassLoader` and we can use it directly here. Apart from the `durationAmount` part we also need to take into account the `timeUnit` part to get an accurate clone. Minimal test:

```
        HazelcastExpiryPolicy policy = new HazelcastExpiryPolicy(3L, 5L, 4L, TimeUnit.SECONDS);
        HazelcastExpiryPolicyConstructor constructor = new HazelcastExpiryPolicyConstructor(HazelcastExpiryPolicy.class);
        HazelcastExpiryPolicy cloned = (HazelcastExpiryPolicy) constructor.createNew(policy);
        assertEquals(policy, cloned);
```

Reply:
Strike two with using JCache classes eagerly where I was not supposed to (I remember one time where I was resolving some classes during node startup :)). Will update the code to what you suggested.

Reply:
Misunderstood the first time but now it's clear. Thanks for the suggestion. Added a test for the constructor as well and applied changes to 3.12.z PR.